modules/detection_and_odometry.py

```diff
@@ -0,0 +1,77 @@
+"""
+LiDAR detection and local odometry data structure.
+"""
+
+from . import drone_odometry_local
+
+
+class LidarDetection:
+    """
+    Lidar scan
+    """
+
+    __create_key = object()
+
+    @classmethod
+    def create(cls, distance: float, angle: float) -> "tuple[bool, LidarDetection | None]":
```

Comment:
we should have a couple checks here, for example, negative distance or angle > limit should return False, None
the angle limits I think should be a class constant